sql/postgres/migrate.go

```diff
@@ -421,6 +421,30 @@ func (m *migrate) fks(b *sqlx.Builder, fks ...*schema.ForeignKey) {
 	})
 }
 
+// skipAutoDropped filters unnecessary changes that are automatically
+// dropped by the database before running ALTER TABLE.
+func skipAutoDropped(changes []schema.Change) []schema.Change {
+	dropC := make(map[string]bool)
+	for _, c := range changes {
+		if c, ok := c.(*schema.DropColumn); ok {
+			dropC[c.C.Name] = true
+		}
+	}
+	for i, c := range changes {
+		switch c := c.(type) {
+		case *schema.DropIndex:
+			if len(c.I.Parts) == 1 && c.I.Parts[0].C != nil && dropC[c.I.Parts[0].C.Name] {
```

Comment:
can you elaborate here a bit in doc? this is not the easiest code to read

Reply:
Done. Thanks 🙏